pkg/prom/instance/group_manager.go

```diff
@@ -328,6 +328,22 @@ func groupConfigs(groupName string, grouped groupedConfigs) (Config, error) {
 	combined.Name = groupName
 	combined.ScrapeConfigs = []*config.ScrapeConfig{}
 
+	// Assign all remote_write configs in the group a consistent set of remote_names.
+	// If the grouped configs are coming from the scraping service, defaults will have
+	// been applied and the remote names will be prefixed with the old instance config name.
+	for _, rwc := range combined.RemoteWrite {
+		// Blank out the existing name before getting the hash so it is doesn't take into
+		// account any existing name.
+		rwc.Name = ""
+
+		hash, err := getHash(rwc)
+		if err != nil {
+			return Config{}, err
+		}
+
+		rwc.Name = groupName[:6] + "-" + hash[:6]
```

Comment:
Out of curiosity, why the first six?

Reply:
It's just a pattern I'm copying from [Prometheus](https://github.com/prometheus/prometheus/blob/f4a1b3cef606b478b8dac0d07c1ebb0a2f47e6ce/storage/remote/write.go#L114-L120). It's highly unlikely to get a collision here with the number of remote write endpoints people tend to have, and trimming it to 13 characters here is much more reasonable than the 33 characters it'd be otherwise.